engine/src/main/java/org/hibernate/validator/internal/engine/cascading/ValueExtractorManager.java

```diff
@@ -31,6 +32,34 @@
 public class ValueExtractorManager {
 
 	@Immutable
+	public static final Set<ValueExtractor<?>> SPEC_DEFINED_EXTRACTORS = Stream.of(
+			ByteArrayValueExtractor.DESCRIPTOR,
+			ShortArrayValueExtractor.DESCRIPTOR,
+			IntArrayValueExtractor.DESCRIPTOR,
+			LongArrayValueExtractor.DESCRIPTOR,
+			FloatArrayValueExtractor.DESCRIPTOR,
+			DoubleArrayValueExtractor.DESCRIPTOR,
+			CharArrayValueExtractor.DESCRIPTOR,
+			BooleanArrayValueExtractor.DESCRIPTOR,
+			ObjectArrayValueExtractor.DESCRIPTOR,
+
+			LegacyListValueExtractor.DESCRIPTOR,
+			ListValueExtractor.DESCRIPTOR,
+
+			LegacyMapValueExtractor.DESCRIPTOR,
+			MapValueExtractor.DESCRIPTOR,
+			MapKeyExtractor.DESCRIPTOR,
+
+			LegacyIterableValueExtractor.DESCRIPTOR,
+			IterableValueExtractor.DESCRIPTOR,
+
+			OptionalValueExtractor.DESCRIPTOR,
+			ObjectValueExtractor.DESCRIPTOR
```

Comment:
Considering how useful `getDefaultValueExtractors` is, I would have made it a Set of VED instead of VE and reuse it in the constructor. That could avoid forgetting to add one at one place or the other.

And we could stream the result directly in `getDefaultValueExtractors`.